### ISSUE TITLE
Confusing scroll bounce effect in iOS

### ISSUE THREAD

**Issue by DILewis** [oerc-music/ForbiddenQuestion#3]
**Problem:** Even though this is a single-page app, with no scrolling, you still get an elastic scroll effect in iOS as you swipe with a finger. This gives the impression that there is more material to see by scrolling (encouraging users to keep trying to scroll).

Online suggestions include:

1. **CSS fixes:** Making containers have fixed position and/or overflow: hidden – **tried and failed**
1. **Event callbacks:** Adding bespoke callbacks to touch events and then using preventDefault() – not yet tried.
1. **Meta tags:** There have been some bespoke meta tags in the past that solved this, but none work now.
1. **Safari-specific CSS:** `-webkit-overflow-scrolling` and others don't work anymore – **tried and failed** (but this is a moving target
1. **Convert to an app:** There are special options in native apps to disable this – **not an option**
1. **Libraries:**
  a. **bouncefix and react-bouncefix:** Don't work. Demo on their site doesn't work. Repo has little or no activity. Seems to use method 1 above.
  a. **iNoBounce:** Demo doesn't work, but this seems more recent. There is dev activity and outstanding PRs on the issues. It appears to use method 2 above. I _think_ from the discussion that the PR won't work, but have not yet tried.

iOS has a history of changing how this can be bypassed with each version. An advantage of using a library (if it works) is that it's more likely to track these changes.

**Comment by gklyne** [comment]
Have you seen/tried this:

overscroll-behavior-y: none;

(found at https://www.bram.us/2016/05/02/prevent-overscroll-bounce-in-ios-mobilesafari-pure-css/)

See also: https://bugs.webkit.org/show_bug.cgi?id=176454 - it looks as if there's a community trying to fix this - work is quite recently active (comment as recent as 2020-09-07).

**Comment by DILewis** [comment]
Looks like it's not (yet?) supported: [caniuse.com](https://caniuse.com/css-overscroll-behavior)

**Comment by gklyne** [comment]
Yes.  I note that the [age I linked mentions other workarounds - I can't tell if you've tried these.  Longer term, I think `overscroll-behavior` would be a preferred option.

**Comment by DILewis** [comment]
I _believe_ that the workarounds described in that first article are the ones I tried under category 1. Should be fairly easy to retest with dev tools and safari.
